packages/service-library/src/servicelib/fastapi/prometheus_instrumentation.py

```diff
@@ -7,18 +7,20 @@
 
 def setup_prometheus_instrumentation(app: FastAPI) -> Instrumentator:
 
-    instrumentator = (
-        Instrumentator(
-            should_instrument_requests_inprogress=True, inprogress_labels=False
-        )
-        .instrument(app)
-        .expose(app, include_in_schema=False)
-    )
-
-    def _unregister():
-        for collector in list(instrumentator.registry._collector_to_names.keys()):
+    instrumentator = Instrumentator(
+        should_instrument_requests_inprogress=True, inprogress_labels=False
+    ).instrument(app)
+
+    async def on_startup(app: FastAPI) -> None:
+        instrumentator.expose(app, include_in_schema=False)
+
+    def _unregister() -> None:
```

Comment:
what about making this one async as well, could it help?

Reply:
not useful